orchagent/main.cpp

```diff
@@ -140,6 +145,121 @@ void init_gearbox_phys(DBConnector *applDb)
     delete tmpGearboxTable;
 }
 
+bool getSystemPortConfigList(vector<sai_system_port_config_t> &sysportcfglist)
+{
+    DBConnector cfgDb("CONFIG_DB", 0);
+    DBConnector appDb("APPL_DB", 0);
+
+    Table cfgDeviceMetaDataTable(&cfgDb, CFG_DEVICE_METADATA_TABLE_NAME);
+    Table cfgSystemPortTable(&cfgDb, CFG_SYSTEM_PORT_TABLE_NAME);
+    Table appSystemPortTable(&appDb, APP_SYSTEM_PORT_TABLE_NAME);
+
+    //At this point of time (i.e, at the time of orchagent start), the CONFIG_DB is completely fully
+    //populated with the contents of config_db.json.
+
+    string value;
+    //Get the swith type.
+    if(!cfgDeviceMetaDataTable.hget("localhost", "switch_type", value))
```

Comment:
Could you move the code lines that get switch_type out of function getSystemPortConfigList()? Inside getSystemPortConfigList(), we could expect gMySwitchType must be "voq".

It is because if gMySwitchType is "fabric", we would do something else. 

Moving these code lines out of this function will help avoiding reading CONFIG_DB multiple times to identify asic type.

So, you could define DB connectors out of this function too, and have them as getSystemPortConfigList arguments.

Reply:
Yes. We'll fix as suggested. 
The DB connection to config db, app db and state db happens at later point of time in this function. We'll move the db connection before getting the switch type and system port config list and pass the db connectors as arguments to get switch type and get sytem port config list functions.

Reply:
Fixed